### DISCUSSION TITLE
view function

### DISCUSSION THREAD

**Discussion by krishno16** [Cyfrin/foundry-full-course-cu#2425] [General]
a view function that can be accessed from children's contracts
can you please tell me how can i write this i can't understand

**Comment by EngrPips** [comment]
Hello @krishno16, A child contract gets access to all the functions of the Parent contract except for private functions. That is the essence of inheritance `to make child contract use functions that weren't explicitly defined in them`

```js
contract Parent {
function add(uint256 a, uint256 b) internal view returns(uint256) {
uint256 Z = a + b;
}
}
```

```js
import {Parent} from "./Parent.sol"

contract Child is Parent {
function calculate(uint256 a, uint256 b) external view returns(uint256) {
uint256 result = add(a, b)
return result;
}
}
```
The above is an example of a child contract accessing the view function of the Parent contract.
